File: src/main/java/no/nav/sosialhjelp/soknad/business/batch/oppgave/Oppgave.java
package no.nav.sosialhjelp.soknad.business.batch.oppgave;

import no.nav.sosialhjelp.soknad.business.batch.oppgave.fiks.FiksData;
import no.nav.sosialhjelp.soknad.business.batch.oppgave.fiks.FiksResultat;
import no.nav.sosialhjelp.soknad.db.repositories.JAXBHelper;

import java.time.LocalDateTime;

public class Oppgave {
    public Long id;
    public String behandlingsId;
    public String type;
    public Status status;
    public int steg;
    public FiksData oppgaveData = new FiksData();
    public FiksResultat oppgaveResultat = new FiksResultat();
    public LocalDateTime opprettet;
    public LocalDateTime sistKjort;
    public LocalDateTime nesteForsok;
    public int retries;

    public void nesteSteg() {
        steg++;
    }

    public void ferdigstill() {
        this.status = Status.FERDIG;
    }


    public enum Status {
        KLAR, UNDER_ARBEID, FERDIG, FEILET
    }

    public final static JAXBHelper JAXB = new JAXBHelper(
            FiksData.class,
            FiksResultat.class
    );


}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/db/repositories/oppgave/Oppgave.kt
package no.nav.sosialhjelp.soknad.db.repositories.oppgave

import com.migesok.jaxb.adapter.javatime.LocalDateTimeXmlAdapter
import no.nav.sosialhjelp.soknad.db.repositories.JAXBHelper
import java.time.LocalDateTime
import javax.xml.bind.annotation.XmlRootElement
import javax.xml.bind.annotation.XmlType
import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter

data class Oppgave(
    var id: Long,
    var behandlingsId: String,
    var type: String?,
    var status: Status,
    var steg: Int,
    var oppgaveData: FiksData? = FiksData(),
    var oppgaveResultat: FiksResultat? = FiksResultat(),
    var opprettet: LocalDateTime?,
    var sistKjort: LocalDateTime?,
    var nesteForsok: LocalDateTime?,
    var retries: Int
) {
    fun nesteSteg() {
        steg++
    }

    fun ferdigstill() {
        status = Status.FERDIG
    }
}

enum class Status {
    KLAR, UNDER_ARBEID, FERDIG, FEILET
}

@XmlRootElement
data class FiksData(
    var behandlingsId: String? = null,
    var avsenderFodselsnummer: String? = null,
    var mottakerOrgNr: String? = null,
    var mottakerNavn: String? = null,
    var dokumentInfoer: List<DokumentInfo>? = null,
    @XmlJavaTypeAdapter(LocalDateTimeXmlAdapter::class)
    var innsendtDato: LocalDateTime? = null,
    var ettersendelsePa: String? = null,
)

@XmlRootElement
@XmlType(name = "fiksDokumentInfo")
data class DokumentInfo(
    var uuid: String?,
    var filnavn: String?,
    var mimetype: String?
)

@XmlRootElement
data class FiksResultat(
    var fiksForsendelsesId: String? = null,
    var feilmelding: String? = null
)

val JAXB = JAXBHelper(
    FiksData::class.java,
    FiksResultat::class.java
)
